lib/JsonClient.js

```diff
@@ -98,6 +105,14 @@ JsonClient.prototype.parse = function parse(req, callback) {
             }
         }
 
+        // if no http error but we had a json parse error, return the json
+        // parse err as the top level error
+        if (!resErr && parseErr) {
+            resErr = new RestError(parseErr, {
+                message: 'Invalid JSON in response'
+            });
+        }
+
         if (resErr) {
             resErr.body = obj;
```

Comment:
In the event of a parse error, `body` would be a good candidate for the invalid JSON string.

Reply:
Yeah, that stems from the legacy issue discussed above. I thought about it some, since we're going to breaking behavior anyway we might as well do it now to keep it strict/consistent with what's valid JSON. I'll make that change.